otelib/backends/services/base.py

```diff
@@ -29,18 +29,35 @@ def __init__(self, source: str) -> None:
         super().__init__(source)
 
         self.url: "Optional[str]" = source
+        self._headers: "Optional[Dict[str, Any]]" = None
         self.settings = Settings()
 
+    @property
+    def headers(self) -> "Dict[str, Any]":
+        """URL headers to use for all requests to the OTEAPI Service."""
+        return self._headers or {}
+
+    @headers.setter
+    def headers(self, value: "Dict[str, Any]") -> None:
+        """Set the URL headers to use for all requests to the OTEAPI Service."""
+        if not isinstance(value, dict):
+            raise TypeError("headers must be a dictionary")
+        self._headers = value
+
     def create(self, **config) -> None:
         session_id = config.pop("session_id", None)
         data = self.strategy_config(**config)
 
+        headers = self.headers or {}
```

Comment:
That is true. But as far as I understood, the playloads in the OTEApi are always of type `application/json`. Hence submitting e.g. a configuration of type `text/turtle` would not resolve in a successful response from the OTE-services side. 

Ergo, why not keep client-side assignment of the `Content-Type` to the `headers` here and add another exception if the user-defined `Content-Type` is invalid?

Reply:
PS: Do we know if the `headers`-object is not transformed by the `requests`-module internally when it is passed to the related `post`/`get`-method called by the OTEClient ?

Reply:
It should never be transformed by the `requests` module.

Also, I like your suggestion above to evaluate and subsequently set the `Content-Type` header. I would do this upon initialization though, i.e., not in the `create` method. There should be some utility method for initializing I think? I.e., one that isn't `__init__()`. If not, you could create one that is called from `__init__()`.

Reply:
@CasperWA, I just added the default `Content-Type` to the `headers`-property, since the default ones are `None` during `__init__()`. If they are returned, `application/json` is injected as `Content-Type`, if not available.

What do you think?

Reply:
All good. I would've preferred a separate issue to follow-up, but this is fine as well.